packages/react-native/scripts/cocoapods/utils.rb

```diff
@@ -189,7 +189,7 @@ def self.add_build_settings_to_pod(installer, settings_name, settings_value, tar
         installer.target_installation_results.pod_target_installation_results.each do |pod_name, target_installation_result|
             if pod_name.to_s == target_pod_name
                 target_installation_result.native_target.build_configurations.each do |config|
-                        if configuration == nil || (configuration != nil && configuration == config.name)
+                        if configuration == nil || (configuration != nil && config.name.ends_with?(configuration))
```

Comment:
Great finding! But what if, instead, we use `include?`?
At least we can also cover the following cases:
* Debug
* StagingDebug
* DebugStaging
* MyDebugConfiguration

What do you think?

```suggestion
                        if configuration == nil || (configuration != nil && config.name.include?(configuration))
```

Reply:
Indeed, thanks for the suggestion !